static/app/views/insights/common/components/releaseSelector.tsx

```diff
@@ -134,8 +160,77 @@ function LabelDetails(props: LabelDetailsProps) {
   );
 }
 
+function getSortRealesesBy(
+  sort: ReleasesSortByOption,
+  environments: string[]
+): ReleasesSortByOption {
+  // Require 1 environment for date adopted
+  if (sort === ReleasesSortOption.ADOPTION && environments.length !== 1) {
+    return ReleasesSortOption.DATE;
+  }
+
+  const sortExists = Object.keys(SORT_BY_OPTIONS).includes(sort ?? '');
+
+  if (sortExists) {
+    return sort;
+  }
+
+  // We could give a visual feedback to the user, saying that the sort by is invalid but
+  // since this UI will be refactored, maybe we just don't do anything now
+  return ReleasesSortOption.DATE;
+}
+
 export function ReleaseComparisonSelector() {
   const {primaryRelease, secondaryRelease} = useReleaseSelection();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const {selection} = usePageFilters();
+
+  const [localStoragedReleaseBy, setLocalStoragedReleaseBy] =
+    useLocalStorageState<ReleasesSortByOption>(
+      'insightsReleasesSortBy',
+      ReleasesSortOption.DATE
+    );
+
+  const urlStoragedReleaseBy = decodeScalar(
+    location.query.sortReleasesBy
+  ) as ReleasesSortByOption;
+
+  useEffect(() => {
+    if (urlStoragedReleaseBy === localStoragedReleaseBy) {
+      return;
+    }
+
+    // this is useful in case the user shares the url with another user
+    // and the user has a different sort by in their local storage
+    if (!urlStoragedReleaseBy) {
+      navigate(
+        {
+          ...location,
+          query: {
+            ...location.query,
+            sortReleasesBy: localStoragedReleaseBy,
+          },
+        },
+        {replace: true}
+      );
+      return;
+    }
+
+    setLocalStoragedReleaseBy(urlStoragedReleaseBy);
+  }, [
```

Comment:
You can just inline this without an effect
```suggestion
    if (!urlStoragedReleaseBy && urlStorageReleaseBy !== localStorageReleaseby) {
      navigate(
        {
          ...location,
          query: {
            ...location.query,
            sortReleasesBy: localStoragedReleaseBy,
          },
        },
        {replace: true}
      );
    }
   
   const storage = urlStorageReleaseBy || localStorageReleaseBy
```

Reply:
hmmm but isn't updating the URL a side effect? side effects should live inside a `useEffect `

Reply:
It is an effect, but that doesn't mean it needs to be inside a useEffect :) What will happens here is that because useEffect is async, the UI might flicker with the old value for a split second, before it gets replaced as the result of the effect run, which I think is something you want to avoid

Reply:
I have a question about this! React documentation is _very explicit_ that renders _must_ be pure (though I guess in this case, it's not a huge deal). Maybe this is a good case for `useLayoutEffect`, since it block browser paint?